examples/chat/module/rust/src/lib.rs

```diff
@@ -14,137 +14,92 @@
+        info!(
+            "handling incoming gRPC invocation with labels: request={:?} response={:?}",
+            grpc_request_label, grpc_response_label
+        );
+        if grpc_request_label.can_flow_to(&grpc_response_label) {
+            // Retrieve a reference to a room for the request label, or create one if it does not
+            // already exist.
+            let room_node_sender = self
+                .rooms
+                .entry(grpc_request_label.clone())
+                .or_insert_with(|| create_room_node(&grpc_request_label));
+            // Forward the incoming gRPC invocation to the appropriate room node.
+            room_node_sender.send(&grpc_invocation)
+        } else {
+            // If the caller is trying to send a message with a label such that it would not be able
+            // to receive it itself, do not even attempt to process the request and instead return
+            // an error immediately.
+            warn!("gRPC request and response have inconsistent labels");
+            let mut grpc_response_writer =
+                oak::grpc::ChannelResponseWriter::new(grpc_invocation.response_sender);
+            grpc_response_writer.close(Err(oak::grpc::build_status(
+                oak::grpc::Code::PERMISSION_DENIED,
+                "inconsistent labels",
+            )))
         }
     }
+}
 
-    fn subscribe(&mut self, req: SubscribeRequest, mut writer: grpc::ChannelResponseWriter) {
-        info!("subscribing to room");
-        match self.rooms.get(&req.room_id) {
-            None => {
-                writer
-                    .close(room_id_not_found_err())
-                    .expect("could not close channel");
-            }
-            Some(room) => {
-                info!("new subscription to room {:?}", req.room_id);
-                let command = Command::Join(writer.handle());
-                room.sender
-                    .send(&command)
-                    .expect("could not send command to room Node");
-            }
-        };
-    }
-
-    fn send_message(&mut self, req: SendMessageRequest) -> grpc::Result<Empty> {
-        info!("sending message to room");
-        match self.rooms.get(&req.room_id) {
-            None => room_id_not_found_err(),
-            Some(room) => {
-                info!("new message to room {:?}", req.room_id);
-                let message_bytes = req
-                    .get_message()
-                    .write_to_bytes()
-                    .expect("could not convert message to bytes");
-                let command = Command::SendMessage(message_bytes);
-                room.sender
-                    .send(&command)
-                    .expect("could not send command to room Node");
-                Ok(Empty::new())
-            }
-        }
-    }
+fn create_room_node(grpc_invocation_label: &oak::label::Label) -> oak::io::Sender<Invocation> {
+    let (wh, rh) = oak::channel_create_with_label(grpc_invocation_label).unwrap();
```

Comment:
Does this channel actually need the `grpc_invocation_label` ?  If only `Invocation`s  (i.e. pairs of handles) are sent down it, then possibly not?

Reply:
You are correct, changed back to plain `channel_create`.